src/dbnode/persist/fs/types.go

```diff
@@ -622,3 +622,6 @@ type Segments interface {
 	AbsoluteFilePaths() []string
 	BlockStart() time.Time
 }
+
+// InfoFileResultsPerShard is a map mapping shards to info files.
+type InfoFileResultsPerShard map[uint32][]ReadInfoFileResult
```

Comment:
nit: you seem to use `map[namespace.Metadata]InfoFileResultsPerShard` a lot, maybe better to define that as a type here for ease of use?